src/Exception/UnexpectedValueException.php

```diff
@@ -1,14 +0,0 @@
-<?php
-
-namespace Http\Client\Exception;
-
-use Http\Client\Exception;
-
-/**
- * Custom Unexpected Value exception
- *
- * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
- */
-final class UnexpectedValueException extends \UnexpectedValueException implements Exception
-{
```

Comment:
This was only used in the BatchException and BatchResult. This is thrown in case of a "programming error" as you've said.